libvast/vast/experimental_table_slice.hpp

```diff
@@ -27,6 +28,9 @@ struct experimental_table_slice_state {
 
   /// The deserialized Arrow Record Batch.
   std::shared_ptr<arrow::RecordBatch> record_batch;
+
+  /// Mapping from column offset to nested Arrow array
+  arrow::ArrayVector array_index;
```

Comment:
The variable names is quite unclear to me. What about `flat_columns`?

Reply:
I'll rename this in the other branch where we merge back into `arrow_table_slice` b/c this file is going away, and all we get is a merge conflict.